File: frontend/src/controllers/Gradebook.java
package controllers;

import helpers.ChangeWindow;
import helpers.DynamicPages;
import javafx.event.ActionEvent;
import javafx.fxml.FXML;
import javafx.scene.Node;
import javafx.scene.Parent;
import javafx.scene.control.Pagination;
import javafx.scene.control.Button;
import javafx.util.Callback;

public class Gradebook {

    @FXML
    private Pagination main_paginator;

    @FXML
    private Button logout;

    @FXML
    void click(ActionEvent event) {
        //TODO delete session
        ChangeWindow changer = new ChangeWindow("/fxml/login.fxml", logout);
        changer.hide();
        changer.load();
    }

    @FXML
    public void initialize() {
        main_paginator.setPageFactory(new Callback<Integer, Node>() {
            private String[] allPages = {
                    "start", "information", "marks", "marks", "marks", "marks", "marks",
                    "marks", "marks", "marks", "marks", "marks", "marks", "marks", "marks",
                    "marks", "marks", "marks", "marks", "marks", "marks", "marks", "marks", "marks",
                    "marks", "marks", "practiceTable", "practice", "atestations", "examsMarks",
                    "degreeProject", "examinationCommission"
            };

            @Override
            public Node call(Integer param) {
                String pageName = allPages[param];
                String path = "/fxml/gradebook/" + pageName + ".fxml";
                return pageName == "marks" ?
                        new DynamicPages(path, param.toString()).loadPage() :
                        new DynamicPages(path).loadPage();
            }
        });
    }
}

File: backendJS/gradebookDesktop/src/API/gradebook/gradebook.js
const {requestCreator} = require("shvidko");

const information = requestCreator("get", "/information", (req, res) => {
    res.send("hello from gradebook")
})

const marks = requestCreator("get", "/marks/?semester;species", (req, res) => {
    res.send("hello from gradebook")
})

const practiceTable = requestCreator("get", "/practiceTable", (req, res) => {
    res.send("hello from gradebook")
})

const practice = requestCreator("get", "/practice", (req, res) => {
    res.send("hello from gradebook")
})

const atestations = requestCreator("get", "/atestations", (req, res) => {
    res.send("hello from gradebook")
})

const examsMarks = requestCreator("get", "/examsMarks", (req, res) => {
    res.send("hello from gradebook")
})

const degreeProject = requestCreator("get", "/degreeProject", (req, res) => {
    res.send("hello from gradebook")
})

const examinationCommission = requestCreator("get", "/examinationCommission", (req, res) => {
    res.send("hello from gradebook")
})

module.exports = [information, atestations, degreeProject, examinationCommission, examsMarks, marks, practice, practiceTable]